src/solvers/refinement/string_constraint_generator_testing.cpp

```diff
@@ -41,7 +43,7 @@ exprt string_constraint_generatort::add_axioms_for_is_prefix(
     qvar,
     prefix.length(),
     isprefix,
-    equal_exprt(str[plus_exprt_with_overflow_check(qvar, offset)],
+    equal_exprt(str[plus_exprt(qvar, offset)],
```

Comment:
Why can you remove the overflow checks, here and twice below?

Reply:
First, we have to remove them from inside string_constraints because the way `plus_exprt_with_overflow_check` are implemented does not make sense when inside a universal formula.
The fact that this is not need for correctness comes from the use of `--string-max-length`: as long at is set to something reasonable, no overflow can happen.
The situation is not ideal now because we should either force setting `--string-max-length` or improve overflow detection, but this is outside the scope of this PR.

To summarize: the code before was incorrect, now it is correct if `--string-max-length` is used.